Mathlib/Topology/UniformSpace/Dini.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Topology.Algebra.UniformGroup.Basic
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Topology.Order.MonotoneConvergence
+import Mathlib.Topology.UniformSpace.CompactConvergence
+
+/-! # Dini's Theorem
+
+This file proves Dini's theorem, which states that if `F n` is a monotone increasing sequence of
+continuous real-valued functions on a compact set `s` converging pointwise to a continuous function
+`f`, then `F n` converges uniformly to `f`.
+
+We generalize the codomain from `ℝ` to a linearly ordered commutative group `G` with the order
+topology which is also order closed. Therefore it applies also when the codomain is `ℤ`, `ℚ`, or
+`ℝ≥0ˣ`.
+-/
+
+open Filter Topology
+
+variable {ι α G : Type*} [SemilatticeSup ι]
+    [Nonempty ι] [TopologicalSpace α] [LinearOrderedCommGroup G] [UniformSpace G]
+    [UniformGroup G] [OrderTopology G] [OrderClosedTopology G] [Nontrivial G]
```

Comment:
Since `G` is linearly ordered, `OrderClosedTopology G` comes for free.

I have thought about removing the linearity assumption, but my mind is not clear yet about it, I'll add more once it is the case.